Unable to use the leggedodometry class

When I try to use the legged odometry I got the following meaningless results
```
[ERROR] Model :: getFrameLink : frameIndex -1 is not valid, should be between 0 and 269
[ERROR] Model :: getFrameTransform : frameIndex -1 is not valid, should be between 0 and 269
[LeggedOdometry::updateKinematics] Initialized LeggedOdometry.
base --------------->     -nan      -nan      -nan 0.0477658  0.391864  0.918559 0.0202494
[2021-04-26 22:21:41.472] [thread: 92728] [blf] [info] COM-------------> -nan -nan -nan
[2021-04-26 22:21:41.472] [thread: 92728] [blf] [info] l_sole-------------> 0.978525 -0.00038232 0.206127
-0.148468 0.692382 0.706091
-0.142989 -0.721531 0.677456
 -nan -nan -nan

[2021-04-26 22:21:41.472] [thread: 92728] [blf] [info] r_sole-------------> 0.978518 -5.98931e-05 0.20616
-0.148805 0.691904 0.706488
-0.142686 -0.721989 0.677032
 -nan -nan -nan
```


The legged odometry is used as follows
```cpp
    if (!m_floatingBaseEstimator.initialize(handler->getGroup("FLOATING_BASE_ESTIMATOR"),
                                            kinDyn.kindyn))
    {
        BipedalLocomotion::log()->error("{} Unable to initialize the legged odometry.", logPrefix);
        return false;
    }

    m_floatingBaseEstimator.changeFixedFrame(contactListMap["left_foot"].defaultIndex(),
                                             contactListMap["left_foot"].firstContact()->pose.quat(),
                                             contactListMap["left_foot"].firstContact()->pose.translation());

    m_floatingBaseEstimator.setKinematics(m_currentJointPos, m_currentJointVel);
    m_floatingBaseEstimator.advance();

    m_baseTransform = m_floatingBaseEstimator.getOutput().basePose;
    m_baseVelocity = m_floatingBaseEstimator.getOutput().baseTwist;
```

The `ini` file contains
```ini
sampling_period_in_s      0.001

[ModelInfo]
base_link                 "root_link"
base_link_imu             "root_link"
left_foot_contact_frame   "l_sole"
right_foot_contact_frame  "r_sole"

[LeggedOdom]
initial_fixed_frame       "l_sole"
switching_pattern         "useExternal"
```
cc @prashanthr05

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -53,6 +53,7 @@ All notable changes to this project are documented in this file.
 - Fixed the installation path of public headers related to perception libraries. (https://github.com/dic-iit/bipedal-locomotion-framework/pull/245)
 - Fixed InstallBasicPackageFiles to avoid the same problem of https://github.com/dic-iit/matio-cpp/pull/41 (https://github.com/dic-iit/bipedal-locomotion-framework/pull/253)
 - Call `positionInterface->setRefSpeeds()` only once when a position reference is set in `YarpRobotControl` (https://github.com/dic-iit/bipedal-locomotion-framework/pull/271)
+- Fix initialization of reference frame for world in `LeggedOdometry` class. (https://github.com/dic-iit/bipedal-locomotion-framework/pull/289)
 
 ## [0.1.0] - 2021-02-22
 ### Added

--- a/src/Estimators/src/LeggedOdometry.cpp
+++ b/src/Estimators/src/LeggedOdometry.cpp
@@ -202,6 +202,7 @@ bool LeggedOdometry::customInitialization(std::weak_ptr<BipedalLocomotion::Param
         "The parameter handler could not find \"initial_ref_frame_for_world \" in the configuration file. Setting \"initial_fixed_frame\" as reference frame for world"
         << std::endl;
         m_pimpl->m_initialRefFrameForWorld = m_pimpl->m_initialFixedFrame;
+        m_pimpl->m_initialRefFrameForWorldIdx = m_pimpl->m_initialFixedFrameIdx;
     }
     else
     {

